main.py

```diff
@@ -98,27 +104,33 @@ async def on_message(message: Message):
         await command_stop()
 
 
-# Take a filename as string and return it formatted nicely
-def format_filename(filename: str):
+# Format a song as text nicely using artist/title tags if available
+# local_filepath is the actual path on disc
+# filename is the filename on Discord
+# artist_fallback is the fallback author value (no fallback if not passed)
+def song_format(local_filepath: str, filename: str, artist_fallback: str = ""):
+    # tag from tag list is valid as an entry
+    def valid_tag(tag):
```

Comment:
another candidate for a docstring conversion. though since this function is inline, we probably don't need to go all out with args/return sections.

```suggestion
    def valid_tag(tag):
        """
        Returns True if a given tag list entry is valid, otherwise False.
        
        A "valid" tag is defined as a str that contains at least one non-whitespace character.
        """
```

Reply:
Because it's just a nested helper function, and because `song_format()`itself is so short, i think a docstring here might be overkill. I think changing it to `is_valid_tag` and just a single comment above it defining a valid tag is enough, though let me know if you disagree